ddtrace/tracer/payload.go

```diff
@@ -45,36 +44,38 @@ var _ io.Reader = (*payload)(nil)
 func newPayload() *payload {
 	p := &payload{
 		header: make([]byte, 8),
+		buf:    bytes.NewBuffer(make([]byte, 0, 512)),
```

Comment:
Why 512? Why not the zero value `bytes.Buffer`?

Reply:
To be honest, this one is arbitrary, as I determined empirically that a typical small trace is on average in the 300-500 bytes range. I found it therefore appropriate to preallocate a small buffer of 512 bytes, to avoided multiple allocations upon calling payload.Read. I can switch back to a "zero" buffer if you prefer.

Reply:
I'd honestly prefer to have the zero buffer `bytes.Buffer` in the struct field (and not as a pointer). We can always run the benchmarks and see if this change has any impact. I guess there's no reason for premature optimizations, right?

Reply:
Done.